driver/image_adjust_saturation_driver.hpp

```diff
@@ -0,0 +1,378 @@
+    }
+}
+
+template <typename Tgpu, typename Tref>
+void mloImageAdjustSaturationRunHost(miopen::TensorDescriptor inputDesc,
+                                     miopen::TensorDescriptor outputDesc,
+                                     const Tgpu* input,
+                                     Tref* output,
+                                     float saturation_factor)
+
+{
+    tensor_view_4d_t input_tv  = get_inner_expanded_4d_tv(inputDesc);
+    tensor_view_4d_t output_tv = get_inner_expanded_4d_tv(outputDesc);
+
+    // temporary view for workspace (basically a contiguous vector with same size as input_tv)
+    std::vector<Tref> workspace = std::vector<Tref>(inputDesc.GetElementSize(), 0);
+    miopen::TensorDescriptor wsDesc =
+        miopen::TensorDescriptor{inputDesc.GetType(), inputDesc.GetLengths()};
+
+    auto ws_tv = get_inner_expanded_4d_tv(wsDesc);
+
+    auto N        = inputDesc.GetElementSize();
+    auto c_stride = input_tv.size[2] * input_tv.size[3];
+    auto n_stride = c_stride * input_tv.size[1];
+
+    float bound = 1.0f;
+
+    RGBToGrayscale(input, workspace.data(), input_tv, ws_tv, N / 3);
+    Blend(input,
+          workspace.data(),
+          output,
+          input_tv,
+          ws_tv,
+          output_tv,
+          n_stride,
+          c_stride,
+          N,
+          saturation_factor,
+          bound);
+}
+
+template <typename Tgpu, typename Tref>
+class ImageAdjustSaturationDriver : public Driver
+{
+public:
+    ImageAdjustSaturationDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&inputTensorDesc);
+        miopenCreateTensorDescriptor(&outputTensorDesc);
+        data_type = miopen_type<Tgpu>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+    int RunBackwardCPU();
+
+    int VerifyForward() override;
+    int VerifyBackward() override;
+    ~ImageAdjustSaturationDriver() override
+    {
+        miopenDestroyTensorDescriptor(inputTensorDesc);
+        miopenDestroyTensorDescriptor(outputTensorDesc);
+    }
+
+private:
+    InputFlags inflags;
+    int forw;
+
+    miopenTensorDescriptor_t inputTensorDesc;
+    miopenTensorDescriptor_t outputTensorDesc;
+
+    std::unique_ptr<GPUMem> input_gpu;
+    std::unique_ptr<GPUMem> output_gpu;
+    std::unique_ptr<GPUMem> workspace_gpu;
+
+    std::vector<Tgpu> input_host;
+    std::vector<Tgpu> output_host;
+    std::vector<Tref> output_ref;
+
+    float saturation_factor;
+};
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustSaturationDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only the forward pass", "int");
+
+    inflags.AddTensorFlag("input", 'I', "1x3x96x96", "Input Tensor Size");
```

Comment:
Please add example to description

Reply:
Same as other drivers. This will be added by the framework itself